cloudinstall/charms/__init__.py

```diff
@@ -364,15 +356,33 @@ def watch_deploy(self):
                 self.ui.status_error_message(msg)
             time.sleep(10)
 
+    def filter_valid_relations(self):
```

Comment:
this function should probably have a docstring.
it's kind of a straightforward function but it isn't totally clear how you intend it to be used, so documenting that would be good